### PR TITLE
PUB-1512 Confirmation screen for list type

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link ###

https://tools.hmcts.net/jira/browse/PUB-1512

### Change description ###

Add confirmation screen for list type subscription.

Image 25-08-2022 at 16 03

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/main/service/subscriptionService.ts [hunk 1]
```diff
@@ -199,7 +204,151 @@ export class SubscriptionService {
     return payload;
   }
 
+  public async configureListTypeForLocationSubscriptions(userId, listType): Promise<boolean> {
+    return await subscriptionRequests.configureListTypeForLocationSubscriptions(
+      this.createListTypeSubscriptionPayload(userId, listType));
+  }
+
+  createListTypeSubscriptionPayload(userId, listType): object {
```
Comment (ChrisS1512) [inline src/main/service/subscriptionService.ts:212]: This method can be marked as private

File: src/main/service/subscriptionService.ts [hunk 2]
```diff
@@ -199,7 +204,151 @@ export class SubscriptionService {
...
+    let listTypeArray;
+    if(listType) {
+      if (!Array.isArray(listType)) {
```
Comment (ChrisS1512) [inline src/main/service/subscriptionService.ts:215]: Can this happen? (should it always be either an array, or blank)

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/service/subscriptionService.ts (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/main/service/subscriptionService.ts:215]: I implemented it in the same way function createSubscriptionPayload already there, but I changed it to private now :)

File: src/main/service/subscriptionService.ts (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/main/service/subscriptionService.ts]: I implemented it in the same way function createSubscriptionPayload already there, but I changed it to private now :)

**Review by ChrisS1512 — APPROVED** [review]

File: src/main/resources/locales/en/subscription-configure-list-confirmed.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "title": "Subscriptions updated",
+  "header": "Subscriptions updated",
+  "subheader": "Your subscription(s) have been updated successfully",
```
Comment (KianKwa) [inline src/main/resources/locales/en/subscription-configure-list-confirmed.json:4]: Can you double check the grammar here. The ticket says "has been" but it is "have been" here :)

File: src/main/resources/locales/en/subscription-configure-list-confirmed.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "p1": "To continue, you can go to ",
+  "link_text": "your account",
+  "p2": " in order to:",
+  "addEmailSubscription": "add a new email subscription",
+  "manageYourCurrentSubscriptions": "manage your current email subscriptions",
+  "manageYourListTypeSubscriptions": "Further refine the list types per court received",
```
Comment (KianKwa) [inline src/main/resources/locales/en/subscription-configure-list-confirmed.json:10]: This is part of the bullet points like the other lines but this starts with an upper case but the other ones start with lower case. Consider making them consistent?

Comment (joshblackmoor) [inline src/main/resources/locales/en/subscription-configure-list-confirmed.json:10]: Just wondering if this should be a capital or not?

The mock up does show it but why are the rest on the page lower case.

Seems like we either need all with capital or none?

**Review by KianKwa — APPROVED** [review]

File: src/main/resources/locales/en/subscription-configure-list-confirmed.json (hunk 4 above)
Comment (junaidiqbalmoj) [reply src/main/resources/locales/en/subscription-configure-list-confirmed.json:10]: As per discussion with Chris C, we do not need to include third bullet point. So, I have removed and and updated the rest of content changes on the page.

Comment (junaidiqbalmoj) [reply src/main/resources/locales/en/subscription-configure-list-confirmed.json:10]: As per discussion with Chris C, we do not need to include third bullet point. So, I have removed and and updated the rest of content changes on the page.

**Review by joshblackmoor — APPROVED** [review]
Looks good!

Just the Welsh that is missing from the changes you've made which I assume will be covered in a future ticket.

**Review by furnivall — DISMISSED** [review]

**Review by joshblackmoor — APPROVED** [review]

**Review by KianKwa — APPROVED** [review]

**Pull request merged**